src/item_group.cpp

```diff
@@ -209,10 +209,12 @@ void Item_modifier::modify( item &new_item ) const
     }
 
     new_item.set_damage( rng( damage.first, damage.second ) );
-    int random_dirt = rng( dirt.first, dirt.second );
-    if( new_item.is_gun() && random_dirt > 0 ) {
-        new_item.set_var( "dirt",  random_dirt );
-        new_item.faults.emplace( "fault_gun_dirt" );
+    if( !new_item.has_flag( "PRIMITIVE_RANGED_WEAPON" ) ) {
```

Comment:
Why have the checks separated across two if statements? Just combine it.

Reply:
I thought about it, but the way it's done now it prevents rolling random_dirt, saving processing power? That's the only reason I separated it out.

Reply:
I guess I could move the check if gun up to the new if clause and leave only the dirt checker in the bottom clause.

Reply:
I think it's negligible enough to not care about it.
At the very least, I'd just move all the checks up except the `dirt > 0` to here.

Reply:
ok